File: Java/AlgoExpert/Strings/Easy/Generate Document.java
import java.util.*;

class Program {

  public boolean generateDocument(String characters, String document) {
    // Write your code here.
		Map<Character, Integer> myMap = new HashMap<>();
		
		for(int i = 0; i < characters.length(); i++){
			if(myMap.containsKey(characters.charAt(i))){
				myMap.put(characters.charAt(i), myMap.get(characters.charAt(i)) + 1);
			}else{
				myMap.put(characters.charAt(i), 1);
			}
		}
				 
		for(int i = 0; i < document.length(); i++){
			if(!myMap.containsKey(document.charAt(i)) || myMap.get(document.charAt(i)) == 0)
				return false;
			else{
				myMap.put(document.charAt(i), myMap.get(document.charAt(i)) - 1);
			}
		}
		
    return true;
  }
}

File: Python/AlgoExpert/Strings/Easy/generateDocument.py
def generateDocument(characters, document):
    # Write your code here.
    char_map = {}
    for c in characters:
        if c in char_map:
            char_map[c] += 1
        else:
            char_map[c] = 1

    for c in document:
        if c not in char_map:
            return False
        else:
            if char_map[c] == 1:
                char_map.pop(c)
            else:
                char_map[c] -= 1
    return True
